### PR TITLE
feat(dist): support pre-load test graph data in docker container

### PR DESCRIPTION
**Pull request by aroundabout**
## Purpose of the PR

feature:  pre-load some data in container(subtask of #840)

## Main Changes

1. Provide the related conf and groovy for user to pre load some data. 
2. Change the start-hugegraph.sh to get the environment variables to decide to pre-load or not.

## Verifying these changes

- [ ] Trivial rework / code cleanup without any test coverage. (No Need)
- [ ] Already covered by existing tests, such as *(please modify tests here)*.
- [x] Need tests and can be verified by github action

## Does this PR potentially affect the following parts?

- [ ]  Nope
- [ ]  Dependencies (add/update license info) 
- [ ]  Modify configurations
- [ ]  The public API
- [x]  Other affects (The docker hub should update, or the user should build the docker image themselves)

## Documentation Status

The doc is still WIP.

- [x]  `Doc - TODO` 
- [x]  `Doc - Done` 
- [ ]  `Doc - No Need`

### CODE REVIEW

File: hugegraph-dist/README.md
Comment (imbajin) [inline hugegraph-dist/README.md:1]: docker compose aimed for multi process

this step mark as WIP & review it later

File: hugegraph-dist/src/assembly/static/scripts/example-preload.groovy
Comment (imbajin) [inline hugegraph-dist/src/assembly/static/scripts/example-preload.groovy:65]: lack one empty line

File: hugegraph-dist/src/assembly/static/conf/gremlin-server-preload.yaml
Comment (imbajin) [inline hugegraph-dist/src/assembly/static/conf/gremlin-server-preload.yaml:79]: if we only modify one line, consider use `sed` before load the file? 

so as the `pre-load` groovy file (could show the differ)

Comment (VGalaxies) [reply hugegraph-dist/src/assembly/static/conf/gremlin-server-preload.yaml:79]: > if we only modify one line, consider use `sed` before load the file?
> 
> so as the `pre-load` groovy file (could show the differ)
> 
> and @VGalaxies any suggestion for it to preload some data?

The original purpose of `example.groovy` was to serve as a sample script for Gremlin-Console. After removing `registerRocksDB` and `serverStarted`, it can be used to import data when the server starts up.

**If the content of this script can be guaranteed to remain unchanged**, deleting the fixed lines in `example.groovy` using `sed` can achieve the expected effect (keep `gremlin-server.yaml` unchanged).

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh
Comment (Radeity) [inline hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:84]: Some tiny suggestion, better use double square brackets to avoid exception if user don't set `PRELOAD`, also, we can add condition check whether the value is true. Like: `if [[ $PRELOAD = "true" ]]; then`.

File: hugegraph-dist/src/assembly/static/conf/gremlin-server-preload.yaml
Comment (aroundabout) [reply hugegraph-dist/src/assembly/static/conf/gremlin-server-preload.yaml:79]: > > if we only modify one line, consider use `sed` before load the file?
> > so as the `pre-load` groovy file (could show the differ)
> > and @VGalaxies any suggestion for it to preload some data?
> 
> The original purpose of `example.groovy` was to serve as a sample script for Gremlin-Console. After removing `registerRocksDB` and `serverStarted`, it can be used to import data when the server starts up.
> 
> So, deleting the corresponding lines in `example.groovy` using `sed` can achieve the expected effect (no need to introduce `gremlin-server-preload.yaml` and `example-preload.groovy`).
> 
> ```shell
> sed -i -e '/registerRocksDB/d; /serverStarted/d' example.groovy
> ```

Thank you! And I have another question. 
If the users want to test the algorithm like PageRank, they need a demo dataset like movieLen(ml-latest-small) with more vertices and edges. 
Should we provide such feature? Or just preload these data by groovy in current version. And can we preload the file data like [hugegraph-loader](https://hugegraph.apache.org/docs/quickstart/hugegraph-loader/#44-command-to-import) in server or hubble?

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh
Comment (aroundabout) [reply hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:84]: thx for your suggestion

Comment (aroundabout) [reply hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:84]: > Some tiny suggestion, better use double square brackets to avoid exception if user don't set `PRELOAD`, also, we can add condition check whether the value is true. Like: `if [[ $PRELOAD = "true" ]]; then`.

thx for your suggestion

File: hugegraph-dist/src/assembly/static/conf/gremlin-server-preload.yaml
Comment (VGalaxies) [reply hugegraph-dist/src/assembly/static/conf/gremlin-server-preload.yaml:79]: > > > if we only modify one line, consider use `sed` before load the file?
> > > so as the `pre-load` groovy file (could show the differ)
> > > and @VGalaxies any suggestion for it to preload some data?
> > 
> > 
> > The original purpose of `example.groovy` was to serve as a sample script for Gremlin-Console. After removing `registerRocksDB` and `serverStarted`, it can be used to import data when the server starts up.
> > So, deleting the corresponding lines in `example.groovy` using `sed` can achieve the expected effect (no need to introduce `gremlin-server-preload.yaml` and `example-preload.groovy`).
> > ```shell
> > sed -i -e '/registerRocksDB/d; /serverStarted/d' example.groovy
> > ```
> 
> Thank you! And I have another question. If the users want to test the algorithm like PageRank, they need a demo dataset like movieLen(ml-latest-small) with more vertices and edges. Should we provide such feature? Or just preload these data by groovy in current version. And can we preload the file data like [hugegraph-loader](https://hugegraph.apache.org/docs/quickstart/hugegraph-loader/#44-command-to-import) in server or hubble?

This is just my personal opinion, but I think it's enough to create a basic graph by preloading this data through the groovy script when server starting. Users can then use Loader or Hubble to import the data they need (like movieLen) 🤔

Comment (aroundabout) [reply hugegraph-dist/src/assembly/static/conf/gremlin-server-preload.yaml:79]: > > > > if we only modify one line, consider use `sed` before load the file?
> > > > so as the `pre-load` groovy file (could show the differ)
> > > > and @VGalaxies any suggestion for it to preload some data?
> > > 
> > > 
> > > The original purpose of `example.groovy` was to serve as a sample script for Gremlin-Console. After removing `registerRocksDB` and `serverStarted`, it can be used to import data when the server starts up.
> > > So, deleting the corresponding lines in `example.groovy` using `sed` can achieve the expected effect (no need to introduce `gremlin-server-preload.yaml` and `example-preload.groovy`).
> > > ```shell
> > > sed -i -e '/registerRocksDB/d; /serverStarted/d' example.groovy
> > > ```
> > 
> > 
> > Thank you! And I have another question. If the users want to test the algorithm like PageRank, they need a demo dataset like movieLen(ml-latest-small) with more vertices and edges. Should we provide such feature? Or just preload these data by groovy in current version. And can we preload the file data like [hugegraph-loader](https://hugegraph.apache.org/docs/quickstart/hugegraph-loader/#44-command-to-import) in server or hubble?
> 
> This is just my personal opinion, but I think it's enough to create a basic graph by preloading this data through the groovy script when server starting. Users can then use Loader or Hubble to import the data they need (like movieLen) 🤔

All right, thx for your opinion.

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh
Comment (javeme) [inline hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:88]: expect just one blank line

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh
Comment (javeme) [inline hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:85]: prefer to make a copy, then modify the copied file

File: hugegraph-dist/README.md
Comment (imbajin) [inline hugegraph-dist/README.md:22]: thanks, we should also provide the doc for `non-docker-compose` user to use `docker` cmd directly? (to load example data)

Comment (aroundabout) [reply hugegraph-dist/README.md:22]: OK, I will supplement it in the doc.

**Review by javeme — APPROVED** [review]
LGTM

**Review by imbajin — DISMISSED** [review]
LGTM, after this PR merged, we should also update the doc in DockerHub page & website (doc repo) [**TODO**]

**Review by coderzc — DISMISSED** [review]

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh
Comment (imbajin) [inline hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:84]: how could someone load the example data directly(without docker?)

seems `-p true` is better? (and we could also support input `$PRELOAD` param)

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh
Comment (imbajin) [inline hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:26]: lack default info:
image

**Review by imbajin — APPROVED** [review]
LGTM, and remember to add the related doc in website

File: hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh
Comment (imbajin) [inline hugegraph-dist/src/assembly/static/bin/start-hugegraph.sh:38]: **update the doc** in website (like `quick-start` section) to tell users (quick load with `-p true` when start server)

**Review by coderzc — APPROVED** [review]

**Pull request merged**
